MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp

```diff
@@ -438,10 +438,19 @@ Show the user the dialog for an algorithm
 void QtReflMainView::showAlgorithmDialog(const std::string &algorithm) {
   std::stringstream pythonSrc;
   pythonSrc << "try:\n";
-  pythonSrc << "  " << algorithm << "Dialog()\n";
+  pythonSrc << "  algm = " << algorithm << "Dialog()\n";
+  if (algorithm == "LoadReflTBL") {
```

Comment:
1. Split this functionality into a new public method called showImportDialog.
2. Don't do the algorithm check. Instead, look at the algorithm properties via getProperties. Look for a key called 'OutputWorkspace' if it exists. Fetch it.